updates for issue#68

John-Kaippallimalil opened the pull request.

John-Kaippallimalil committed:
updates for issue#68

sridharangirish91 commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
Are we delving into inter-flow priorities, outside of the same UDP 4-tuple? I thought we mindfully wanted to move away due to a lot of complexities.

sridharangirish91 commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
 # Use Cases {#uc}
 
-## Media Streaming {#uc-streaming}
+## Streaming and Interactive Media {#uc-media}
```
IMHO, Even if the requirements are overlapping, Streaming and Interactive Media should be separate items, given they are 2 big use cases that comprise of most of the internet traffic. On the other hand, we can merge interactive media, accommodate delay and mixed traffic into one.

sridharangirish91 commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
+REQ-PACKET-DELAY: Metadata to indicate whether the packet can tolerate delay.
 
-Accommodate-Delay traffic includes content that is more resilient to buffering and delays (e.g., file copy, file download) compared to interactive traffic. This traffic can be buffered in favor of improved interactivity, without significant impact to the user experience. It is least sensitive to buffer bloating.
+## Mixed-Traffic and User Preferences {#uc-preferences}
```
User Preference might not belong here given that is agnostic of the traffic type but is an end user option. I am not sure if combining these 2 makes sense

sridharangirish91 commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -546,64 +472,15 @@ Taking into account these challenges, and despite the use case is described in t
-  1. Performance Optimization: Some applications support some forms of bandwidth measurements (e.g., {{app-measurement}}) which feed how the content is accessed to using ABR. Complementing or replacing these measurements with explicit signals will improve overall network performance and can help optimize the data transfer. Signaling bandwidth availability allows hosts to avoid contributing to net ...
-
-  2. Dynamic Adaptation: When the network informs the host about available bandwidth, the host can dynamically adjust its data transmission rate.
+### Delay Tolerance between Flows (Inter-flow) {#interflow-delay}
```
Again, we should refrain from any inter-flow discussions IMO.

sridharangirish91 committed:
reworking the doc to make it more focused. Moving some parts to extended section in the appendix

sridharangirish91 committed:
rewording host to more specific terminology

danwing committed:
merely did a fix-lint

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
 {: #Figure-netshaper title=”Metadata and Network Shaping”}
 
-{{Figure-netshaper}} provides an outline of the interaction of application packets with network layer entities. Application layer signaling and feedback between client – server (A – in figure) adjusts rate over a period of several RTTs using feedback and congestion control algorithms.
+{{Figure-netshaper}} shows a bottleneck (access) Router on the path of packets from Server to Client.
```
```suggestion
{{Figure-netshaper}} shows a bottleneck (access) router on the path of packets from Server to Client.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -173,12 +173,16 @@ In this document, this is termed 'reactive policy'.
 
-{{Figure-netshaper}} provides an outline of the interaction of application packets with network layer entities. Application layer signaling and feedback between client – server (A – in figure) adjusts rate over a period of several RTTs using feedback and congestion control algorithms.
+{{Figure-netshaper}} shows a bottleneck (access) Router on the path of packets from Server to Client.
+A network shaper in the Router manages QoS of flows of multiple users and can buffer (delay), discard or apply other flow control rules. Application layer signaling and feedback between client – server (A – in figure) adjusts rate over a period of several RTTs using feedback and congestion control algorithms.
```
```suggestion
A network shaper in the router manages QoS of flows of multiple users and can buffer (delay), discard, or apply other flow control rules. Application layer signaling and feedback between Client – Server (A – in the figure) adjust transmission rate over a period of several RTTs using feedback and congestion control algorithms.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```suggestion
Network shapers observe flows and apply policies to maximize performance but are not aware whether  there is a high preference for one flow (UDP 4-tuple) over another flow belonging to the same user and network attachment (e.g., a subscriber connection, a 3GPP PDU Session. See {{net-attach}} for more details).
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```suggestion
Clients may provide an (access) router with preferences of which flow (UDP 4-tuple) has relative priority among flows of that network attachment using H2N (C - in Figure).
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```suggestion
Clients may also provide information to an (access) router to drop, when resources are congested,  'lower priority'-marked packets of a flow (UDP 4-tuple) temporarily which can in turn allocate bandwidth to other flows of that same network attachment.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
+## Streaming and Interactive Media {#uc-media}
 
-Streaming video contains the occasional key frame ("I-frame") containing a full video frame.
+Streaming media packets carry audio or video between server and client.
```
```suggestion
Streaming media packets carry audio or video between servers and clients.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
-
-  1. A remote desktop user prioritizes graphics update over an on-going file copy operation.
-  2. User application in foreground is given priority over downloading software updates.
+REQ-PACKET-PRIORITY: Packet priority relative to other packets in the transport flow (UDP 4-tuple).
```
```suggestion
REQ-PACKET-PRIORITY: Packet priority relative to other packets within the same transport flow (UDP 4-tuple).
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
-  1. A remote desktop user prioritizes graphics update over an on-going file copy operation.
-  2. User application in foreground is given priority over downloading software updates.
+REQ-PACKET-PRIORITY: Packet priority relative to other packets in the transport flow (UDP 4-tuple).
+REQ-PACKET-DELAY: Metadata to indicate whether the packet can tolerate delay.
```
```suggestion
REQ-PACKET-DELAY: Metadata to indicate whether a packet can tolerate delay.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -206,121 +210,50 @@ Traffic shaping:
-A game or VoIP application may want to signal different metadata for the same type of packet in each direction.
-For example, for a game, video in the server-to-client direction might be more important than audio, whereas input devices (e.g., keystrokes) might be more important than audio.  Each user can have varied preferences for the same type of data originating from the server.
+Some content is more resilient to buffering and delays (e.g., file copy, file download) compared to interactive traffic. This traffic can be buffered in favor of improved interactivity, without significant impact to the user experience.
+For example, an on-going file copy operation flow (UDP 4-tuple) a remote desktop user can tolerate more delay than the flow carrying graphics updates.
```
```suggestion
For example, an on-going file copy operation flow (UDP 4-tuple) of a remote desktop user can tolerate more delay than the flow carrying graphics updates.
```

John-Kaippallimalil committed:
Update draft-kwbdgrr-tsvwg-net-collab-rqmts.md

Revised section 5.2 to add that for REQ-PACKET-PRIORITY the behavior is to drop packets of lower priority (i.e., they are near real-time and cannot tolerate delay)
Added 5.2.2 with REQ-PACKET-DELAY for packets that can tolerate some delay as in LE PHB RFC 8622.

danwing committed:
Apply suggestions from code review

Med's suggested changes

Co-authored-by: Med <mohamed.boucadair@orange.com>

sridharangirish91 committed:
Apply suggestions from code review

Co-authored-by: Med <mohamed.boucadair@orange.com>

sridharangirish91 committed:
nit

sridharangirish91 committed:
IncorrectConflictResolution correction

sridharangirish91 committed:
fix build break

sridharangirish91 committed:
Merge pull request #70 from SpencerDawkins/sridharanr-68

reworking the doc to make it more focused. Moving some parts to exten…

danwing commented:
Incorporated via #70.
